tests/Litestar/test_litestar.py

```diff
@@ -0,0 +1,1127 @@
+    assert info["refreshTokenFromAny"] is not None
+
+    headers: Dict[str, Any] = {}
+    cookies: Dict[str, Any] = {}
+
+    if token_transfer_method == "header":
+        headers.update({"authorization": f"Bearer {info['refreshTokenFromAny']}"})
+    else:
+        cookies.update(
+            {"sRefreshToken": info["refreshTokenFromAny"], "sIdRefreshToken": "asdf"}
+        )
+
+    if info["antiCsrf"] is not None:
+        headers.update({"anti-csrf": info["antiCsrf"]})
+
+    res = driver_config_client.post(
+        "/auth/session/refresh", headers=headers, cookies=cookies
+    )
+    info = extract_info(res)
+
+    assert res.status_code == 401
+    assert_info_clears_tokens(info, token_transfer_method)
+
+
+@mark.asyncio
+@mark.parametrize("token_transfer_method", ["cookie", "header"])
+async def test_revoking_session_after_create_new_session_with_throwing_unauthorized_error(
+    driver_config_client: TestClient[Litestar], token_transfer_method: str
+):
+    init(
+        **get_st_init_args(
+            url=get_new_core_app_url(),
+            recipe_list=[
+                session.init(
+                    anti_csrf="VIA_TOKEN",
+                )
+            ],
+        )
+    )
+
+    res = driver_config_client.post(
+        "/create-throw", headers={"st-auth-mode": token_transfer_method}
+    )
+    info = extract_info(res)
+
+    assert res.status_code == 401
+    assert_info_clears_tokens(info, token_transfer_method)
+
+
+@mark.asyncio
+async def test_session_with_legacy_refresh_token_and_unauthorized_should_clear_legacy_token(
+    driver_config_client: TestClient[Litestar],
+):
+    init(
+        **get_st_init_args(
+            url=get_new_core_app_url(),
+            recipe_list=[
+                session.init(
+                    anti_csrf="VIA_TOKEN",
+                )
+            ],
+        )
+    )
+
+    headers: Dict[str, Any] = {}
+    cookies: Dict[str, Any] = {}
+
+    cookies.update(
+        # Invalid refresh token so that core throws an unauthorized error
+        {"sRefreshToken": "non-existing-token", "sIdRefreshToken": "irrelevant-value"}
+    )
+
+    res = driver_config_client.post(
+        "/auth/session/refresh", headers=headers, cookies=cookies
+    )
+    info = extract_info(res)
+
+    assert res.status_code == 401
+    assert res.json() == {"message": "unauthorised"}
+    assert_info_clears_tokens(info, "cookie")
+
+    assert info["sIdRefreshToken"]["value"] == ""
+    assert info["sIdRefreshToken"]["expires"] == "Thu, 01 Jan 1970 00:00:00 GMT"
+
+
+@mark.asyncio
+async def test_search_with_email_t(driver_config_client: TestClient[Litestar]):
+    init(
+        supertokens_config=SupertokensConfig(get_new_core_app_url()),
+        app_info=InputAppInfo(
+            app_name="SuperTokens Demo",
+            api_domain="http://api.supertokens.io",
+            website_domain="http://supertokens.io",
+            api_base_path="/auth",
+        ),
+        framework="flask",
```

Comment:
```suggestion
        framework="litestar",
```
Please fix all similar instances